codalab/lib/interactive_session.py

```diff
@@ -136,14 +142,23 @@ def get_docker_path(sub_path):
                 volumes[get_docker_path(key)] = dependency_local_path
 
         name = self._get_container_name()
-        command = ['docker run', '-it', f'--name {name}', f'-w {os.path.sep}{self._session_uuid}']
+        # Start a container as a non-root user. Root (id = 0) is the default user within a container.
+        # When passing a numeric ID, the user does not have to exist in the container.
+        command = [
+            'docker run',
+            '-it',
+            f'--name {name}',
+            f'-w {os.path.sep}{self._session_uuid}',
+            '-u 1',
```

Comment:
I'm not too familiar with UIDs, but why pick 1 (daemon) as the UID?

[This page](https://refspecs.linuxbase.org/LSB_3.1.1/LSB-Core-generic/LSB-Core-generic/usernames.html#FTN.AEN26413) says: "The daemon User ID/Group ID was used as an unprivileged User ID/Group ID for daemons to execute under in order to limit their access to the system. Generally daemons should now run under individual User ID/Group IDs in order to further partition daemons from one another."

Reply:
0 is for root and any non-zero number is for a non-root user. You can specify a number without having to register a user.